### PR TITLE
Fix languageId sent for react syntax and update scope

### PR DESCRIPTION
**Pull request by rchl**
"source.js.react" scope is used by JS Custom.
That whole language block needs to be first on the list as it's more
specific than "source.js". Otherwise we would send "languageId: javascript"
for react files.

### CODE REVIEW

**Comment by rchl** [comment]
> You could also move the typescriptreact entry upwards, and add `source.ts.react` for future-proofing :)

How do I know what the name of the typescript react syntax will be? :)
(at least for st3 it's important)

**Comment by rchl** [comment]
Added `source.ts.react` to `Packages/TypeScript Syntax/TypeScriptReact.tmLanguage` syntax but that one will probably not add such scope. It's the "JS Custom" one that likely will have that scope but I don't know what the name of it will be in the future.

**Comment by predragnikolic** [comment]
JSCustom added typescript support pretty recently :)
```
{
    "configurations": {
        "Default": {},
        "React": {
            "file_extensions": [ "js", "jsx" ],
            "flow_types": true,
            "jsx": true,
        },
        "TypeScript": {
            "file_extensions": [ "ts" ],
            "typescript": true,
        },
        "TypeScriptReact": {  // I added this manually
            "file_extensions": [ "tsx" ],
            "typescript": true,
            "jsx": true
        }
    }
}
```

Apparently JSCustom build syntax scopes by the name of the configuration.
For `TypeScript` config the scope is `source.js.typescript`.
For `TypeScriptReact` `source.js.typescriptreact`.

Just checked `TypeScript Syntax`. The scope is `source.tsx` for `.tsx` files, and `source.ts` for `.ts` files.

File: LSP-typescript.sublime-settings [hunk 1]
```diff
@@ -2,40 +2,31 @@
...
 				"Packages/Babel/JavaScript (Babel).sublime-syntax",
 			],
 		},
 		{
-			"languageId": "typescript",
-			"scopes": [
-				"source.ts",
-			],
+			"languageId": "typescriptreact",
+			"scopes": ["source.tsx", "source.ts.react"],
```
Comment (predragnikolic) [inline LSP-typescript.sublime-settings:20]: `source.ts.react` 
Where did you get this scope? :)

Comment (rchl) [reply LSP-typescript.sublime-settings:20]: made it up. Seems consistent with source.js.react

**Comment by rchl** [comment]
> Apparently JSCustom build syntax scopes by the name of the configuration.

Then we can come up with something and require users of lsp-typescript to set that.
For example "TSX" or "TypeScriptReact". I suppose name of the configuration also affects scope so for those two cases it will be "source.ts.tsx" and "source.ts.typescriptreact" respecitively.

**Comment by rchl** [comment]
Removed imagined scopes and syntaxes for now. We'll add it when it exits.

**Pull request merged**